### DISCUSSION TITLE
`vue-mini` + `tailwindcss` 集成示例模板

### DISCUSSION THREAD

**Discussion by sonofmagic** [vue-mini/vue-mini#92] [General]
之前有开发者有需求，集成 vue-mini 和 tailwindcss，于是就改了一份模板，详见：

## 工具链为 `weapp-vite` 版本的模板:

https://github.com/sonofmagic/vue-mini-tailwindcss-template

## 工具链为 `create-vue-mini` 版本的模板

https://github.com/hgaogao/vue-mini-weapp-tailwindcss-template

**Comment by yyqxjwxy** [comment]
good job!

**Comment by Lucascuibu** [comment]
您好，我用这个模版写页面没问题，但是写组件tailwind就不生效。路径也包括在tailwindconfig.js里了，是还需要额外的设置吗

**Reply by sonofmagic** [reply]
微信小程序默认组件样式隔离的，需要声明  apply-shared

可以参考这个

https://developers.weixin.qq.com/miniprogram/dev/framework/custom-component/wxml-wxss.html#%E7%BB%84%E4%BB%B6%E6%A0%B7%E5%BC%8F%E9%9A%94%E7%A6%BB

**Comment by archcst** [comment]
https://github.com/sonofmagic/vue-mini-tailwindcss-template 使用 tdesign-miniprogram 后出现了一些问题

复现步骤：

1. 下载本仓库源码
2. pnpm i
3. pnpm i tdesign-miniprogram -S --production
5. pnpm dev
6. 添加按钮：
index.json
```
{
  "$schema": "https://vite.icebreaker.top/page.json",
  "usingComponents": {
    "t-button": "tdesign-miniprogram/button/button"
  }
}
```

index.wxml
```
<t-button theme="primary">按钮</t-button>
```

7. 按钮无样式，console报错：

<img width="1251" alt="image" src="https://github.com/user-attachments/assets/32ae9fdd-a3ca-4d0c-a41a-eaf0916da9ab">

我试了下，`create-vue-mini`出来的项目是可以的

**Reply by sonofmagic** [reply]
` tdesign-miniprogram` 是需要开启 ES5 转义的，我这个模板默认是不开启的，你开启之后就正常了
![image](https://github.com/user-attachments/assets/d9c9513f-f247-43e2-8641-4a4c70c98b3a)
![image](https://github.com/user-attachments/assets/a448f6f5-4e52-4822-b678-a217ac029d4d)
